slack-lint-checks/src/main/java/slack/lint/DaggerIssuesDetector.kt

```diff
@@ -0,0 +1,274 @@
+                )
+                return
+              }
+              isBinds && containingClass.isInterface -> {
+                // Cannot have a default impl in interfaces
+                if (node.uastBody != null) {
+                  context.report(
+                    ISSUE_BINDS_MUST_BE_ABSTRACT,
+                    context.getLocation(node.uastBody),
+                    ISSUE_BINDS_MUST_BE_ABSTRACT.getBriefDescription(TextFormat.TEXT),
+                  )
+                  return
+                }
+              }
+              containingClass.classKind == JvmClassKind.CLASS -> {
+                val isAbstract = context.evaluator.isAbstract(node)
+                // Binds must be abstract
+                if (isBinds && !isAbstract) {
+                  context.report(
+                    ISSUE_BINDS_MUST_BE_ABSTRACT,
+                    context.getLocation(node),
+                    ISSUE_BINDS_MUST_BE_ABSTRACT.getBriefDescription(TextFormat.TEXT),
+                  )
+                  return
+                } else if (isProvides && isAbstract) {
+                  context.report(
+                    ISSUE_PROVIDES_CANNOT_BE_ABSTRACT,
+                    context.getLocation(node),
+                    ISSUE_PROVIDES_CANNOT_BE_ABSTRACT.getBriefDescription(TextFormat.TEXT),
+                  )
+                  return
+                }
+              }
+              containingClass.classKind == JvmClassKind.INTERFACE && isProvides -> {
+                context.report(
+                  ISSUE_PROVIDES_CANNOT_BE_ABSTRACT,
+                  context.getLocation(node),
+                  ISSUE_PROVIDES_CANNOT_BE_ABSTRACT.getBriefDescription(TextFormat.TEXT),
+                )
+                return
+              }
+            }
+          }
+
+          if (isBinds) {
+            if (node.uastParameters.size != 1) {
+              val locationToHighlight =
+                if (node.uastParameters.isEmpty()) {
+                  node
+                } else {
+                  node.parameterList
+                }
+              context.report(
+                ISSUE_BINDS_WRONG_PARAMETER_COUNT,
+                context.getLocation(locationToHighlight),
+                ISSUE_BINDS_WRONG_PARAMETER_COUNT.getBriefDescription(TextFormat.TEXT)
+              )
+              return
+            }
+          }
+
+          val returnType =
+            node.returnType?.takeUnless {
+              it == PsiTypes.voidType() ||
+                context.evaluator.getTypeClass(it)?.qualifiedName == "kotlin.Unit"
+            }
+              ?: run {
+                // Report missing return type
+                val nodeLocation = node.returnTypeElement ?: node
+                context.report(
+                  ISSUE_RETURN_TYPE,
+                  context.getLocation(nodeLocation),
+                  ISSUE_RETURN_TYPE.getBriefDescription(TextFormat.TEXT),
+                )
+                return
+              }
```

Comment:
nit: there's a lot going on here and the lines right above it. I'd replace the `run` with a regular conditional just to improve the readability

Reply:
0e69398